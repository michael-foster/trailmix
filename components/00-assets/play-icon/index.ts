export { PlayIcon } from './play-icon';
