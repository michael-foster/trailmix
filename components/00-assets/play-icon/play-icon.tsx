import React from 'react';
import { SVG } from '@outside-interactive/web-microfrontends.helpers.types';

export const PlayIcon = ({
  classes,
  fillColor,
  strokeColor,
  strokeWidth
}: SVG) => (
  <svg
    className={classes}
    fill="none"
    role="img"
    viewBox="0 0 34 34"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14 24L22 17L14 10L14 24Z"
      fill={fillColor}
      stroke={strokeColor}
      strokeWidth={strokeWidth}
    />
  </svg>
);
