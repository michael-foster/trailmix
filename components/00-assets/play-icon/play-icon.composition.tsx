import React from 'react';
import { PlayIcon } from './play-icon';

export const BasicPlayIcon = () => <PlayIcon fillColor="white" />;
