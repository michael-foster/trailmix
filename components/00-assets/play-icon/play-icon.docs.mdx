---
description: 'A PlayIcon component.'
labels: ['SVG', 'ASSET']
---

import { PlayIcon } from './play-icon';

## React Component for rendering a play icon SVG

### Component usage

```js
<PlayIcon fillColor="#d2a940" />
```

### Using props to customize the icon

Modify the fill color to see it change live:

```js live
<PlayIcon fillColor="#d2a940" />
```
