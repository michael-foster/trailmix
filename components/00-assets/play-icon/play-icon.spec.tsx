import React from 'react';
import { render } from '@testing-library/react';
import { BasicPlayIcon } from './play-icon.composition';

describe('PlayIcon', () => {
  test('snapshot', () => {
    const { asFragment } = render(<BasicPlayIcon />);

    expect(asFragment()).toMatchSnapshot();
  });
});
