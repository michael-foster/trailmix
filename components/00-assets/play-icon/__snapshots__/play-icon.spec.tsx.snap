// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayIcon snapshot 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    role="img"
    viewBox="0 0 34 34"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14 24L22 17L14 10L14 24Z"
      fill="white"
    />
  </svg>
</DocumentFragment>
`;
